### ISSUE TITLE
Find keep-fnames work-around

### ISSUE THREAD

**Issue by Jefftopia** [Gmousse/dataframe-js#26]
keep-fnames will apply to all classes. If a developer is working in a large code-base, this could add a ton of bloat to their minified output.

I haven't been able to look more into this issue yet, I hope to when I get time. One strategy I wonder about is simply adding a string property to each class identifying it rather than relying on the constructor name.

As always, I'm open to other thoughts.

For reference, the reason this is important for me is that when I apply nested groups to a dataframe, I need to recusrively traverse it, meaning I must distinguish a DataFrame from a GroupedDataFrame. While I'm handling that via `on: string[]`, it's inelegant, and something within the lib itself is failing.

**Comment by Gmousse** [comment]
Hi @Jefftopia,
Indeed keep-fnames was a bit tricky and was added to keep checktypes decorators working with user defined types.
I suggest to remove keep-fnames and try to find an alternative (other library) to these checkTypes decorators (or just remove them if NODE_ENV=production).

However, to check the identity of an instance (DataFrame vs GroupedDataFrame), it's possible to use:
`(new DataFrame()) instanceof DataFrame`
`(new DataFrame()).constructor == DataFrame`

Is that what you are looking for?

**Comment by Jefftopia** [comment]
@Gmousse Thank you for the tip and for acknowledging interest in enhancing the developer experience here.

I have a couple thoughts.

First, The checkTypes decorator is nifty, but do we really need run-time type checking? I think that for javascript either the developer should know the API, which you've documented, or, we should simply create a TypeScript definition file. I develop in TS and would find this valuable regardless of the decorator.

Second, I discovered that Uglify has a mangle options object that allows you to mangle everything except specified class names.

I'm using below, which mangles all except DataFrame and GroupedDataFrame, which resolves my issue.
```
    new UglifyJsPlugin({
      "uglifyOptions": {
        "mangle": {
          "reserved": ["DataFrame", "GroupedDataFrame"]
        }
    }),
```

**Comment by Gmousse** [comment]
> The checkTypes decorator is nifty, but do we really need run-time type checking?

No. I m agree, it's not required. I will see to replace it in a next version. (It was planned but I had fixed this issue with keep-fnames due to a lack of personal time). I will try to see how to resolve this issue this week-end.

> Uglify has a mangle options object

Oh, thanks for the trick. I will use it for my next projects. However, it will be useless to use this option if I remove checkTypes decorators (my main option here).

Thanks for the discussion, stay tuned for the fix (I will give you the feature branch for a feedback in the next 7 days).

**Comment by Gmousse** [comment]
@Jefftopia Be free to test if keepfname is no longer a problem on the linked pr.

I will integrate the code in a larger feature in some days.

**Comment by Jefftopia** [comment]
@Gmousse I will try to look over this soon, tonight or tomorrow. Busy holiday week in the US (Thanksgiving). Thank you. It looks good.

**Comment by Jefftopia** [comment]
@Gmousse Can you propagate these changes to the lib directory so I can be more confident in testing?

**Comment by Jefftopia** [comment]
@Gmousse Can you merge the PR?

**Comment by Gmousse** [comment]
@Jefftopia Uh sorry I was really busy (and I forgot this PR :D).
It will be effective today.

**Comment by Gmousse** [comment]
Merged in 1.2.5 (published). 
Thanks for your issue. Have fun
